azure-iot-device/azure/iot/device/iothub/auth/x509_authentication_provider.py

```diff
@@ -17,14 +17,16 @@ class X509AuthenticationProvider(AuthenticationProvider):
     for device identities connecting directly to an Azure IoT hub.
     """
 
-    def __init__(self, hostname, device_id, x509):
+    def __init__(self, hostname, x509, device_id, module_id=None):
         """
         Constructor for X509 Authentication Provider
         :param hostname: The hostname of the Azure IoT hub.
-        :param device_id: The device unique identifier as it exists in the Azure IoT Hub device registry.
         :param x509: The X509 object containing certificate, key and passphrase
+        :param device_id: The device unique identifier as it exists in the Azure IoT Hub device registry.
+        :param module_id: The module unique identifier of the device. Ir is optional when dealing with only devices.
         """
-        logger.info("Using X509 authentication for {%s, %s}", hostname, device_id)
+        # logger.info("Using X509 authentication for {%s, %s}", hostname, device_id)
+        logger.info("Using X509 authentication for {hostname},{device_id},{module_id}".format(hostname=hostname, device_id=device_id, module_id=module_id))
```

Comment:
Wait, why are you passing in the module id if the only change is that now we log it?

Nothing else is even happening with the module id...